main/src/ca/uwaterloo/flix/language/phase/Stratifier.scala

```diff
@@ -152,6 +156,11 @@ object Stratifier {
       val e2 = visitExp(exp2)
       Expr.LetRec(sym, ann, mod, e1, e2, tpe, eff, loc)
 
+    case Expr.LocalDef(ann, sym, fparams, declaredType, declaredEff, exp1, exp2, tpe, eff, loc) =>
```

Comment:
yeah I think the double types and effects should go away via Desugar

Reply:
Yeah true this is mistake. Just forgot to delete the old `tpe` and `eff` from the AST node